### ISSUE TITLE
Terminal state is not preserved on SIGINT

### ISSUE THREAD

**Issue by Mulling** [tauri-apps/create-tauri-app#509]
If you press `ctrl-c` during any of the prompt type menus the cursor remains hidden. This has been discussed in console-rs/dialoguer/issues/77 and is up to the user to set the SIGINT handler.

I've played around some options to fix this, all are complicated by the overuse of `.unwrawp()` in `lib.rs`. Something along those lines should work.

```rust
use rustix::{
    stdio,
    termios::{self, OptionalActions},
};
use std::{env::args_os, ffi::OsStr, path::Path};

fn main() {
    if let Ok(termios) = termios::tcgetattr(stdio::stderr()) {
        let _ = ctrlc::set_handler(move || {
            if let Err(err) = termios::tcsetattr(stdio::stderr(), OptionalActions::Now, &termios) {
                println!("fail to reset terminal state {}", err)
            }
            eprintln!("\x1b[?25h");
        });
    }
```

And the for each prompt we need to `exit()` instead of unwrapping. Like this for the laguage selecting one:
```rust
        // If only one category is detected, skip prompt
        if categories.len() == 1 {
            Some(categories[0])
        } else if let Ok(index) = Select::with_theme(&ColorfulTheme::default())
            .with_prompt("Choose which language to use for your frontend")
            .items(&categories)
            .default(0)
            .interact()
        {
            Some(categories[index])
        } else {
            exit(1);
        }
```

The use of `tcsetattr` is optional, but good practice for cli applications. And we need to write to stderr, since, for some reason, dialoguer performs all interaction on stderr.

I can submit a PR if this is ok.

**Comment by amrbashir** [comment]
I wouldn't like using `exit`, instead you can propage errors in `try_run` method and catch them inside `run` where you can set your reset code. And a PR is always welcome btw.

**Comment by Mulling** [comment]
I've made a draft at #511, I've tested all the paths and it seems to work, tho I'm not sure on some of the error messages.

**Comment by Mulling** [comment]
Fixed in #511
